eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/PostRestartHtlcCleaner.scala

```diff
@@ -151,7 +151,23 @@ class PostRestartHtlcCleaner(nodeParams: NodeParams, register: ActorRef, initial
           // instead spread across multiple local origins) so we can now forget this origin.
           Metrics.PendingRelayedOut.decrement()
           context become main(brokenHtlcs.copy(relayedOut = brokenHtlcs.relayedOut - origin))
-        case Origin.TrampolineRelayed(origins, _) =>
+        case Origin.ChannelRelayedCold(originChannelId, originHtlcId, _, _) =>
+          if (!brokenHtlcs.settledUpstream.contains(origin)) {
+            log.info(s"received preimage for paymentHash=${fulfilledHtlc.paymentHash}: fulfilling 1 HTLC upstream")
+            PendingRelayDb.safeSend(register, nodeParams.db.pendingRelay, originChannelId, CMD_FULFILL_HTLC(originHtlcId, paymentPreimage, commit = true))
+          }
+          val relayedOut1 = relayedOut diff Set((fulfilledHtlc.channelId, fulfilledHtlc.id))
+          if (relayedOut1.isEmpty) {
+            log.info(s"payment with paymentHash=${fulfilledHtlc.paymentHash} successfully relayed")
+            // We could emit a TrampolinePaymentRelayed event but that requires more book-keeping on incoming HTLCs.
+            // It seems low priority so isn't done at the moment but can be added when we feel we need it.
```

Comment:
In that case I think we can (and should) emit a `ChannelPaymentRelayed`

Reply:
Done with 6be5ba134.